lib/modules/platform/azure/index.ts

```diff
@@ -990,3 +994,42 @@ export async function deleteLabel(
   const azureApiGit = await azureApi.gitApi();
   await azureApiGit.deletePullRequestLabels(config.repoId, prNumber, label);
 }
+
+function cachePr(pr?: GitPullRequest): void {
+  config.prList ??= [];
+  // istanbul ignore if
+  if (!pr) {
+    return;
+  }
+
+  const azurePr = getRenovatePRFormat(pr);
+  const idx = getCachedPrIdx(azurePr.number);
+  if (is.undefined(idx)) {
+    config.prList.push(azurePr);
+  } else {
+    config.prList[idx] = azurePr;
+  }
+}
+
+function updateCachedPr(prNo: number, pr?: GitPullRequest): void {
+  config.prList ??= [];
+  const idx = getCachedPrIdx(prNo);
+  if (!pr || is.undefined(idx)) {
+    // if pr doesn't exist in cache then invalidate cache so we get new list from api next time.
+    config.prList = undefined as any;
```

Comment:
```suggestion
    delete config.prList;
```
much simpler, also works if not defined

Reply:
Skill issue on my side xD

Reply:
ah, i can't do it. this conflicts with your previous review comment, prList is not optional.
`TS2790: The operand of a delete operator must be optional`